### PR TITLE
Design/update error

### PR DESCRIPTION
**Pull request by nicolasbarb**

### CODE REVIEW

File: easyselling/Additions/UI/Components/ImagedBackground.swift [hunk 1]
```diff
@@ -16,6 +16,8 @@ struct ImagedBackground<Content : View>: View {
 
     var body: some View {
         ZStack {
+            Color.backgroundColor
```
Comment (MaxenceMottard) [inline easyselling/Additions/UI/Components/ImagedBackground.swift:19]: Maybe use `.background` modifier ?

File: easyselling/Additions/Extensions/View+Extensions.swift [hunk 2]
```diff
@@ -20,4 +20,8 @@ extension View {
     func fillMaxSize(alignment: Alignment = .center) -> some View {
         self.frame(maxWidth: .infinity, maxHeight: .infinity, alignment: alignment)
     }
+
+    func ableToShowError(_ error: String, when isShowingError: Bool) -> some View {
+        modifier(ErrorShower(error: error, isShowingError: isShowingError))
+    }
```
Comment (MaxenceMottard) [inline easyselling/Additions/Extensions/View+Extensions.swift]:
```suggestion

    func ableToShowError(_ error: Error?) -> some View {
        modifier(ErrorShower(error: Error?))
    }
```

File: easyselling/Additions/UI/Components/ErrorShower.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+//
+//  Created by Nicolas Barbosa on 16/12/2021.
+//
+
+import SwiftUI
+
+struct ErrorShower: ViewModifier {
+    @Namespace private var animation
+    var error: String
+    var isShowingError: Bool
```
Comment (MaxenceMottard) [inline easyselling/Additions/UI/Components/ErrorShower.swift]:
```suggestion
    var error: Error?
```

File: easyselling/Additions/UI/Components/ErrorShower.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                            .font(.headline)
+                            .foregroundColor(.white)
+                            .padding(25)
+                    }
+                    .frame(maxWidth: .infinity)
+                    .background(Color.red)
+                    .cornerRadius(30)
+                    .padding(.top, 100)
+                    .transition(.move(edge: .top))
+                }
```
Comment (MaxenceMottard) [inline easyselling/Additions/UI/Components/ErrorShower.swift]:
```suggestion
                if let error = error {
                    VStack {
                        Text(error.errorDescription)
                            .font(.headline)
                            .foregroundColor(.white)
                            .padding(25)
                    }
                    .frame(maxWidth: .infinity)
                    .background(Color.red)
                    .cornerRadius(30)
                    .padding(.top, 100)
                    .transition(.move(edge: .top))
                }
```

File: easyselling/Scenario/AuthenticationScenario/AccountCreation/AccountCreationView.swift [hunk 5]
```diff
@@ -16,15 +16,13 @@ struct AccountCreationView: View {
             ScrollView(showsIndicators: false) {
                 accountCreationView
             }
+            .ableToShowError(viewModel.error?.errorDescription ?? "", when: viewModel.error == nil ? false : true)
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AuthenticationScenario/AccountCreation/AccountCreationView.swift]:
```suggestion
            .ableToShowError(viewModel.error)
```

File: easyselling/Scenario/AuthenticationScenario/PasswordReset/Request/PasswordResetRequestView.swift [hunk 6]
```diff
@@ -50,6 +46,8 @@ struct PasswordResetRequestView: View {
             .ignoresSafeArea(.keyboard, edges: .bottom)
         }
         .padding(25)
+        .ableToShowError(viewModel.error?.errorDescription ?? "",
+                         when: viewModel.error == nil ? false : true)
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AuthenticationScenario/PasswordReset/Request/PasswordResetRequestView.swift]:
```suggestion
        .ableToShowError(viewModel.error)
```

File: easyselling/Scenario/AuthenticationScenario/PasswordReset/Request/PasswordResetRequestViewModel.swift [hunk 7]
```diff
@@ -8,7 +8,7 @@
 import Foundation
 import SwiftUI
 
-class PasswordResetRequestViewModel: ObservableObject {
+class PasswordResetRequestViewModel: NSObject, ObservableObject {
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AuthenticationScenario/PasswordReset/Request/PasswordResetRequestViewModel.swift]: Really necessary ?

File: easyselling/Scenario/AuthenticationScenario/PasswordReset/Reset/PasswordResetView.swift [hunk 8]
```diff
@@ -54,6 +49,7 @@ struct PasswordResetView: View {
             .ignoresSafeArea(.keyboard, edges: .bottom)
         }
         .padding(25)
+        .ableToShowError(viewModel.error?.errorDescription ?? "", when: viewModel.state == .error)
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AuthenticationScenario/PasswordReset/Reset/PasswordResetView.swift]:
```suggestion
        .ableToShowError(viewModel.error)
```

File: easyselling/Scenario/AuthenticationScenario/UserAuthentication/UserAuthenticationView.swift [hunk 9]
```diff
@@ -19,6 +19,7 @@ struct UserAuthenticationView: View {
             ScrollView(.vertical, showsIndicators: false) {
                 loginView
             }
+            .ableToShowError(viewModel.error?.errorDescription ?? "", when: viewModel.error == nil ? false : true)
```
Comment (MaxenceMottard) [inline easyselling/Scenario/AuthenticationScenario/UserAuthentication/UserAuthenticationView.swift]:
```suggestion
            .ableToShowError(viewModel.error)
```

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

File: easyselling/Scenario/AuthenticationScenario/PasswordReset/Request/PasswordResetRequestViewModel.swift (hunk 7 above)
Comment (nicolasbarb) [reply easyselling/Scenario/AuthenticationScenario/PasswordReset/Request/PasswordResetRequestViewModel.swift]: i'm not sure

**Review by MaxenceMottard — DISMISSED** [review]

**Review by MaxenceMottard — APPROVED** [review]

**Pull request merged**